### PR TITLE
[VEN-2604][VEN-2597]: add vip to transfer funds

### PR DESCRIPTION
**Pull request by GitGuru7**
## Description

Resolves VEN-2604, VEN-2597

### CODE REVIEW

File: vips/vip-304/bscmainnet.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+export const REQUIRED_VUSDC_AMOUNT_FOR_REMAINING_USDC = parseUnits("14046500", 8);
+export const USDC_AMOUNT_ON_TREASURY = parseUnits("16777", 18);
+export const REMAINING_USDC_AMOUNT = parseUnits("334623", 18);
+
+export const BNB_TREASURY = "0xf322942f644a996a617bd29c16bd7d231d9f35e9";
+export const COMMUNITY_WALLET = "0xc444949e0054A23c44Fc45789738bdF64aed2391";
+
+export const ETH_AMOUNT_COMMUNITY_WALLET = parseUnits("1.188225443", 18);
+export const USDT_AMOUNT_COMMUNITY_WALLET = parseUnits("10518.73", 18);
+export const BNB_AMOUNT_COMMUNITY_WALLET = parseUnits("3", 18);
```
Comment (chechu) [inline vips/vip-304/bscmainnet.ts:32]: These funds shouldn't be transferred to the Community wallet, but to the DEV_WALLET

Comment (GitGuru7) [reply vips/vip-304/bscmainnet.ts]: [c4535f2](https://github.com/VenusProtocol/vips/pull/288/commits/c4535f23411cec1c7434eb6be958d3e6c96c72b1)

File: vips/vip-304/bscmainnet.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+export const BTC = "0x7130d2A12B9BCbFAe4f2634d864A1Ee1Ce3Ead9c";
+export const vUSDC = "0xecA88125a5ADbe82614ffC12D0DB554E2e2867C8";
+export const TOKEN_REDEEMER = "0x2367bff367Bc4a83A65EEEe3b9E4D834Ff4f637e";
+
+export const WBNB_AMOUNT_DEV_FUND = parseUnits("1472.197705207413945", 18);
+export const ETH_AMOUNT_DEV_FUND = parseUnits("62.069755574301455", 18);
+export const BTC_AMOUNT_DEV_FUND = parseUnits("1.702705080868696", 18);
+export const USDT_AMOUNT_DEV_FUND = parseUnits("21100", 18);
+
+export const REQUIRED_VUSDC_AMOUNT_FOR_REMAINING_USDC = parseUnits("14046500", 8);
```
[the remark this answers is not in the document]
Comment (chechu) [reply vips/vip-304/bscmainnet.ts:23]: In `redeemUnderlyingAndTransfer` only the specified amount of the underlying token is withdrawn from the vToken contract. So, there shouldn't be a surplus of underlying tokens in the TokenRedeemer contract after the execution. There will be a surplus of vTokens in the TokenRedeemer, that will be send to the `receiver` address (the Venus Treasury in this case)

File: vips/vip-304/bscmainnet.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+const { bscmainnet } = NETWORK_ADDRESSES;
+
+export const DEV_WALLET = "0x48e9d2128321cbf75cd108321459865357c00f15";
+export const WBNB = "0xbb4CdB9CBd36B01bD1cBaEBF2De08d9173bc095c";
+export const USDC = "0x8AC76a51cc950d9822D68b83fE1Ad97B32Cd580d";
+export const USDT = "0x55d398326f99059fF775485246999027B3197955";
+export const ETH = "0x2170Ed0880ac9A755fd29B2688956BD959F933F8";
+export const BTC = "0x7130d2A12B9BCbFAe4f2634d864A1Ee1Ce3Ead9c";
+export const vUSDC = "0xecA88125a5ADbe82614ffC12D0DB554E2e2867C8";
+export const TOKEN_REDEEMER = "0x2367bff367Bc4a83A65EEEe3b9E4D834Ff4f637e";
```
Comment (chechu) [inline vips/vip-304/bscmainnet.ts:16]: We can use `0xC53ffda840B51068C64b2E052a5715043f634bcd`, the latest `TokenRedeemer` contract

Comment (GitGuru7) [reply vips/vip-304/bscmainnet.ts]: [bdc98c2](https://github.com/VenusProtocol/vips/pull/288/commits/bdc98c26846b4b3c99e22ab6ec9c846a118d0f8e)

File: vips/vip-304/bscmainnet.ts [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+export const WBNB_AMOUNT_DEV_FUND = parseUnits("1472.197705207413945", 18);
+export const ETH_AMOUNT_DEV_FUND = parseUnits("62.069755574301455", 18);
+export const BTC_AMOUNT_DEV_FUND = parseUnits("1.702705080868696", 18);
+export const USDT_AMOUNT_DEV_FUND = parseUnits("21100", 18);
+
+export const REQUIRED_VUSDC_AMOUNT_FOR_REMAINING_USDC = parseUnits("14046500", 8);
+export const USDC_AMOUNT_ON_TREASURY = parseUnits("16777", 18);
+export const REMAINING_USDC_AMOUNT = parseUnits("334623", 18);
+
+export const BNB_TREASURY = "0xf322942f644a996a617bd29c16bd7d231d9f35e9";
```
[the remark this answers is not in the document]
Comment (GitGuru7) [reply vips/vip-304/bscmainnet.ts]: [bdc98c2](https://github.com/VenusProtocol/vips/pull/288/commits/bdc98c26846b4b3c99e22ab6ec9c846a118d0f8e)

[the remark this answers is not in the document]
Comment (GitGuru7) [reply vips/vip-304/bscmainnet.ts]: [bdc98c2](https://github.com/VenusProtocol/vips/pull/288/commits/bdc98c26846b4b3c99e22ab6ec9c846a118d0f8e)

**Review by chechu — APPROVED** [review]

File: vips/vip-307/bscmainnet.ts [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+export const DEV_WALLET = "0x48e9d2128321cbf75cd108321459865357c00f15";
+export const WBNB = "0xbb4CdB9CBd36B01bD1cBaEBF2De08d9173bc095c";
+export const USDC = "0x8AC76a51cc950d9822D68b83fE1Ad97B32Cd580d";
+export const ETH = "0x2170Ed0880ac9A755fd29B2688956BD959F933F8";
+export const BTC = "0x7130d2A12B9BCbFAe4f2634d864A1Ee1Ce3Ead9c";
+export const vUSDC = "0xecA88125a5ADbe82614ffC12D0DB554E2e2867C8";
+export const TOKEN_REDEEMER = "0xC53ffda840B51068C64b2E052a5715043f634bcd";
+
+export const WBNB_AMOUNT_DEV_FUND = parseUnits("1346.029696578437702", 18); // $834,400, assuming 1 BNB = $619.60
```
Comment (kkirka) [inline vips/vip-307/bscmainnet.ts:17]: `1346.029696578437702 * 619.60` is $834,000, not $834,400

Comment (chechu) [reply vips/vip-307/bscmainnet.ts:17]:
```suggestion
export const WBNB_AMOUNT_DEV_FUND = parseUnits("1346.675274370561653", 18); // $834,400, assuming 1 BNB = $619.60
```

**Comment by chechu** [comment]
https://app.venus.io/#/governance/proposal/307?chainId=56

**Pull request merged**